kafkauser_controller.go:143: segmentation violation

**Describe the bug**
Running master branch, i deployed the operator + simple sample.
I created a topic "my-topic" and i tried to create a KafkaUser producer

``` 
➜  samples git:(master) ✗ k logs -f kafka-operator-operator-5fcccc959b-pk24b -n kafka manager
{"level":"info","ts":1573569677.5931847,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1573569677.5937414,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5939991,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5940893,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.59417,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5942564,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5943215,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.594341,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.594406,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.594428,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5944517,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5944676,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkacluster","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5945892,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkatopic","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5947204,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkauser","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5948384,"logger":"controller-runtime.controller","msg":"Starting EventSource","controller":"kafkauser","source":"kind source: /, Kind="}
{"level":"info","ts":1573569677.5949306,"logger":"setup","msg":"starting manager"}
{"level":"info","ts":1573569677.5953622,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1573569678.8963401,"logger":"controller-runtime.certwatcher","msg":"Updated current TLS certificate"}
{"level":"info","ts":1573569678.896584,"logger":"controller-runtime.certwatcher","msg":"Starting certificate watcher"}
{"level":"info","ts":1573569693.7067783,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"kafkauser"}
{"level":"info","ts":1573569693.7074153,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"kafkacluster"}
{"level":"info","ts":1573569693.7074645,"logger":"controller-runtime.controller","msg":"Starting Controller","controller":"kafkatopic"}
{"level":"info","ts":1573569693.8070407,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"kafkauser","worker count":1}
{"level":"info","ts":1573569693.8073084,"logger":"controllers.KafkaUser","msg":"Reconciling KafkaUser","kafkauser":"kafka/example-producer","Request.Name":"example-producer"}
E1112 14:41:33.807537       1 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:76
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:65
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:679
/usr/local/go/src/runtime/panic.go:199
/usr/local/go/src/runtime/signal_unix.go:394
/workspace/pkg/pki/pki_manager.go:36
/workspace/controllers/kafkauser_controller.go:143
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:216
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192
/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1357
{"level":"info","ts":1573569693.8076081,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"kafkacluster","worker count":1}
{"level":"info","ts":1573569693.8076656,"logger":"controller-runtime.controller","msg":"Starting workers","controller":"kafkatopic","worker count":1}
{"level":"info","ts":1573569693.8077004,"logger":"controllers.KafkaCluster","msg":"Reconciling KafkaCluster","kafkacluster":"kafka/kafka","Request.Name":"kafka"}
{"level":"info","ts":1573569693.807784,"logger":"controllers.KafkaTopic","msg":"Reconciling KafkaTopic","kafkatopic":"kafka/kafka-cruise-control-topic","Request.Name":"kafka-cruise-control-topic"}
{"level":"info","ts":1573569693.80783,"logger":"controllers.KafkaTopic","msg":"Retrieving Kafka client for kafka/kafka","kafkatopic":"kafka/kafka-cruise-control-topic","Request.Name":"kafka-cruise-control-topic"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x130917d]

goroutine 590 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/runtime/runtime.go:58 +0x105
panic(0x181fdc0, 0x29c8920)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/banzaicloud/kafka-operator/pkg/pki.GetPKIManager(0x1d0df60, 0xc00021eab0, 0xc0001ad000, 0xc00090cd19, 0x5)
	/workspace/pkg/pki/pki_manager.go:36 +0x2d
github.com/banzaicloud/kafka-operator/controllers.(*KafkaUserReconciler).Reconcile(0xc00042baa0, 0xc00090cd19, 0x5, 0xc00090cd00, 0x10, 0xc000849c00, 0x0, 0x0, 0x0)
	/workspace/controllers/kafkauser_controller.go:143 +0x5cd
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00030e500, 0x1891500, 0xc00000d4a0, 0x453200)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:216 +0x162
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00030e500, 0x696400)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc00030e500)
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000a87b40)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000a87b40, 0x3b9aca00, 0x0, 0xc000c0b801, 0xc000607080)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000a87b40, 0x3b9aca00, 0xc000607080)
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:157 +0x32e
➜  samples git:(master) ✗
```


**Steps to reproduce the issue:**
Please describe the steps to reproduce the issue.
``` 
cat << EOF | kubectl apply -n kafka -f -
apiVersion: kafka.banzaicloud.io/v1alpha1
kind: KafkaUser
metadata:
  name: example-producer
  namespace: kafka
spec:
  clusterRef:
    name: kafka
  secretName: example-producer-secret
  topicGrants:
    - topicName: my-topic
      accessType: write
EOF
```

--- a/controllers/kafkauser_controller.go
+++ b/controllers/kafkauser_controller.go
@@ -133,6 +133,11 @@ func (r *KafkaUserReconciler) Reconcile(request reconcile.Request) (reconcile.Re
 		}
 		return requeueWithError(reqLogger, "failed to lookup referenced cluster", err)
 	}
+	// Avoid panic if the user wants to create a kafka user but the cluster is in plaintext mode
+	// TODO: refactor this and use webhook to validate if the cluster is eligible to create a kafka user
+	if cluster.Spec.ListenersConfig.SSLSecrets == nil {
+		return requeueWithError(reqLogger, "could not create kafka user since cluster does not use ssl", errors.New("failed to create kafka user"))
+	}
 
 	pkiManager := pki.GetPKIManager(r.Client, cluster)
 

